Update diagrams and release-process description (#380)
alphagov/paas-team-manual: source/team/working_practices.html.md

7. The team move the card to the 'Done' column of the Pivotal Tracker and a team member with appropriate access signs off the work.

![Sequence of events promoting a release](/diagrams/release-sequence.svg)

[Full size version](/diagrams/release-sequence.svg)

The release process consists of the following steps:

1. Developers (engineers) commit code to a development branch on a git repository.

2. When the development is complete, a developer raises a pull request against the main branch of the git repository for review.

> 3. If the review is passed, the reviewer merges the changes into the main branch.

Reviewer: ```suggestion
3. Another team member reviews the pull request and if there are no problems, merges the changes into the main branch.
```

Author: Resolved/ outdated - see below

> 3. If the review is passed, the reviewer merges the changes into the main branch.

Reviewer: Is it worth also stating what happens if there is an issue with something in the PR?

Author: Resolving below

> 3. Another team member reviews the pull request and if there are no problems, merges the changes into the main branch.

Author: ```suggestion
3. Another team member reviews the pull request and if there are no problems, merges the changes into the main branch. Alternatively, they may also provide feedback to the developers and request corrections and/or additional work.
```

Author: Agreed on wording on Skype

> 4. A new commit to the main branch triggers a Concourse git_resource that perforns an automated deployment to the staging environment.

Reviewer: ```suggestion
4. A new commit to the main branch triggers a Concourse git_resource that performs an automated deployment to the staging environment.
```

5. The staging environment runs automated tests.
  - 5 a. If the deployment or tests fail, the automated release process is halted and the team is notified.
  - 5 b. If the deployment is successful and all tests pass, the main branch commit is tagged for release to production.

6. A new release tag triggers a Concourse git_resource that performs an automated deployment to the production environment.

This process does not include the use of git commit signatures which we use to

> verify developer and merger identity for the paas-cf and paas-bootstrap git

Author: ```suggestion
verify developer and merger identity for a number of git
```

Author: Not all, yet, but more than two
